src/Expr.cc

```diff
@@ -960,11 +960,47 @@ ValPtr BinaryExpr::Fold(Val* v1, Val* v2) const
 			DO_UINT_FOLD(^);
 			break;
 		case EXPR_LSHIFT:
-			DO_INT_FOLD(<<);
+			{
+			if ( is_integral )
+				{
+				if ( i2 < 0 )
+					RuntimeError("left shift by negative value");
```

Comment:
Any reason not to use `RuntimeErrorWithCallStack` for the new error cases?

Reply:
@rsmmr Isn't that for internal issues?  The surrounding code only uses it when there's confusion over a type tag or such.

Reply:
Hmm, yeah, I think you're right. I guess I would have expected an internal error then for such type confusion if the assumption is that it can't happen other than for bugs.